### ISSUE TITLE
_select() storing query information in adapter object

### ISSUE THREAD

**Issue by stephenrauch** [web2py/pydal#303]
In ```adapters/base/_select()```, query specific information is being stored in the adapter. If the sql returned by ```_select()``` is later used after another select with a different table or columns, this construct seems like it would fail.

```python
def _select(self, query, fields, attributes):
   ...
   self._colnames = list(map(colexpand, fields))
```

Don't know the best way to fix this since ```_select()``` currently returns just a string. But could change it to return something like:

```python
class SqlString(str):
    def colnames(self, colnames=None):
        if colnames is not None:
            self._colnames = colnames
        return self._colnames
```

If this approach seems satisfactory, you can assign this to me, if not satisfactory please discuss...

**Comment by gi0baro** [comment]
@niphlod @mdipierro what do you think? I agree with @stephenrauch proposal.

**Comment by niphlod** [comment]
I'm missing the usecase scenario but it seems it wouldn't break existing apps.

**Comment by stephenrauch** [comment]
From the book:

```python
lazy = db(db.person.name=='Jonathan').nested_select(db.person.id)
db(db.thing.id==1).update(owner_id = lazy)
```

If, for some reason, another select was run between these two calls, or if ```lazy``` was re-used later, the update would be broken.

I found the potential problem as I was looking to see if the various uses of _select() can be added to the mongo driver.  So, I have seen no definitive problem from the construct, it is just looked semantically odd to be storing query information in the adapter.

**Comment by niphlod** [comment]
LOL. I honestly didn't know about nested_select() until now. Now I get the usecase.

**Comment by mdipierro** [comment]
I do not oppose to this but this seems work that would benefit very few use cases. So Stephen, it you want to do it, I will add it.

**Comment by gi0baro** [comment]
@stephenrauch this should be fixed by #332 in current master. I'm gonna close this, feel free to re-open it if you still experiencing problems.

**Comment by stephenrauch** [comment]
Thanks for taking care of this, and great work on the refactor.  This was
never a problem for me, it was just something I ran across when looking at
the code.
